compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -221,14 +221,8 @@ declare_lint! {
     ///
     /// ### Explanation
     ///
-    /// This lint detects code that is very likely incorrect. When possible,
-    /// the compiler will attempt to detect situations where code can be
-    /// evaluated at compile-time to generate more efficient code. While
-    /// evaluating such code, if it detects that the code will unconditionally
-    /// panic, this usually indicates that it is doing something incorrectly.
-    /// If this lint is allowed, then the code will not be evaluated at
-    /// compile-time, and instead continue to generate code to evaluate at
-    /// runtime, which may panic during runtime.
+    /// This lint detects code that is very likely incorrect because it will always panic, e.g.
+    /// division by zero and out-of-bounds array accesses.
```

Comment:
```suggestion
    /// division by zero and out-of-bounds array accesses. Consider adjusting your code or,
    /// in case the panic is intended, using the `panic!` or `unreachable!` macro instead.
```
do you think this would be helpful here to be make the explanation a bit more helpful?
Mostly locked at the `arithmetic_overflow` lint for reference

Reply:
Sure; what about this?

Reply:
Note that the only time I have actually encountered this in the wild, adjusting the code was not an option, but allowing the lint was the right way. This is in performance sensitive code that should only have debug assertions if the main crate has debug assertions on, but the code was in `core`: https://github.com/rust-lang/rust/blob/master/library/core/src/iter/range.rs#L204-L208

Reply:
That's just a hack because we have this "overflow check inherited from root crate" but we don't have the same for `debug_assert!`, right?

Reply:
yea, that's indeed a hack for this. Maybe we should crate an intrinsic replacement for `cfg(debug_assertions)` that has the same inheriting behaviour